platform/mbed_application.c

```diff
@@ -23,11 +23,33 @@
 
 #if defined(__CORTEX_A9)
 
+static void powerdown_gic(void);
+
 void mbed_start_application(uintptr_t address)
 {
+    __disable_irq();
```

Comment:
Use critical section instead please

Reply:
@c1728p9 Should be fine to be used here?

Reply:
@0xc0170 @c1728p9 
With respect to `__disable_irq`, it is used in this form at Cortex-M too. [Ref](https://github.com/TomoYamanaka/mbed-os/blob/improve_bootloader/platform/mbed_application.c#L53)
Should I change to critical section both?

Reply:
Good catch, I was not aware. That will need to be addressed then separately if @c1728p9 agrees